modules/sdk-coin-avaxp/src/lib/transaction.ts

```diff
@@ -258,14 +258,37 @@ export class Transaction extends BaseTransaction {
   /** @inheritdoc */
   explainTransaction(): TransactionExplanation {
     const txJson = this.toJson();
-    const displayOrder = ['id', 'outputAmount', 'changeAmount', 'outputs', 'changeOutputs', 'fee', 'type'];
+    const displayOrder = [
+      'id',
+      'outputAmount',
+      'changeAmount',
+      'outputs',
+      'changeOutputs',
+      'rewardAddresses',
+      'fee',
+      'type',
+    ];
+
+    let totalIn = new BN(0);
+    this._utxos.map((utxo) => {
+      totalIn = totalIn.add(new BN(utxo.amount));
+    });
+    const outputAmount = txJson.outputs.reduce((p, n) => p.add(new BN(n.value)), new BN(0)).toString();
+    const changeAmount = totalIn.sub(new BN(outputAmount)).toString();
+
     return {
       displayOrder,
       id: txJson.id,
       outputs: txJson.outputs.map((o) => ({ address: o.address, amount: o.value, memo: txJson.memo })),
-      outputAmount: txJson.outputs.reduce((p, n) => p.add(new BN(n.value)), new BN(0)).toString(),
-      changeOutputs: [], // account based does not use change outputs
-      changeAmount: '0', // account base does not make change
+      outputAmount,
+      changeOutputs: [
+        {
+          address: this.fromAddresses.toString(),
```

Comment:
Isn't `fromAddresses` an array? Do we need to figure out which `fromAddress` received the change output(s)?

Reply:
The type for `changeOutputs` in  [`ITransactionRecipient `](https://github.com/BitGo/BitGoJS/blob/master/modules/sdk-core/src/bitgo/baseCoin/iBaseCoin.ts#L66) is a `string` - changing the type of this address from `string` to `string | string[]` has cascading effects throughout  the SDK in other coins. When I tried to override the type for `changeOutputs`, I could not build the SDK due to errors. Instead of not extending `ITransactionExplanation`, I thought to just make the addresses into a single string and then parse it later. Is this a bad idea?

The alternative is to effectively reimplement the `ITransactionExplanation` with that single parameter changed.

Reply:
There is no `changeOutputs` property in `ITransactionRecipient` - nor would this make sense since a recipient does not have a change output. I'm guessing you probably meant the `changeOutputs` property on `ITransactionExplanation`?

Reply:
Yeah - sorry my explanation was bad.

The `changeOutputs` property in `ITransactionExplanation` is the type `ITransactionRecipient[]`, and the `address` property of `ITransactionRecipient` is a `string` - changing the address property type from a `string` to `string | string[]` would mess up a lot of other coins.

Reply:
I understand that, but that's not what I'm asking about.

Each element in `changeOutputs` should correspond to exactly one output. We currently have a list of `fromAddresses` that's getting concatenated into one `address` property in one element in `changeOutputs`. If we have a list of more than one change outputs on the transaction, why aren't we creating separate elements in `changeOutputs` where each corresponds to one address?

If there is only one change output (and thus we should only have one element in `changeOutputs`), then why do we have a list of addresses associated with that change output instead of a single address?

Reply:
Multisig wallet in Avalanche P and X Chains don't have unique address. They use the addresses of the signers. So those 3 addresses are one wallet.